### PR TITLE
Lora Mask based on lora index

### PR DESCRIPTION
**Pull request by hlahkar**
Changes the filling of lora mask from lora_id to lora_index. This is needed to ensure that the mask does not fail in case lora id is greater than max_loras

### CODE REVIEW

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -1832,6 +1789,60 @@ def _check_config(self, batch_size, seq_len, is_prompt, warmup_mode):
...
+                              dtype=self.lora_config.lora_dtype)
+            for i in range(len(lora_ids)):
+                if lora_ids[i] == 0:
+                    continue
+                lora_index = self.lora_manager._adapter_manager.lora_index_to_id.index(lora_ids[i])
+                start_pos = lora_index * self.lora_config.max_lora_rank
+                end_pos = start_pos + self.lora_config.max_lora_rank
+                lora_mask[i, start_pos:end_pos] = ones
+            lora_mask = lora_mask.to('hpu')
+            lora_logits_mask = lora_mask
```
Comment (michalkuligowski) [inline vllm/worker/habana_model_runner.py:1842]: Could you explain why now logits mask is pointing to mask, before it was left None for decode phase.
Is that because now in line 1906 there can be reference to an object that is None? If so I would rather add there check if lora_logits_mask is not none instead of changing the logic of what lora_logits_mask is keeping

Comment (hlahkar) [reply vllm/worker/habana_model_runner.py]: @michalkuligowski for decode phase lora_mask and logits_mask are same; previously we were setting it at execute_model; just moved to create_lora_mask for a cleaner code

File: vllm/worker/habana_model_runner.py [hunk 2]
```diff
@@ -425,6 +421,7 @@ class ModelInputForHPU(ModelRunnerInputBase):
     real_batch_size: Optional[int] = None
     batch_size_padded: Optional[int] = None
     virtual_engine: int = 0
+    lora_ids: Optional[List[int]] = None
     lora_mask: Optional[torch.Tensor] = None
```
Comment (vivekgoe) [inline vllm/worker/habana_model_runner.py:425]: lora_mask & lora_logits_mask can be deleted here, right?

File: vllm/worker/habana_model_runner.py [hunk 3]
```diff
@@ -1866,11 +1865,19 @@ def execute_model(
...
 
+        lora_mask: torch.Tensor = None
+        lora_logits_mask: torch.Tensor = None
         if self.lora_config:
             assert model_input.lora_requests is not None
             assert model_input.lora_mapping is not None
+            assert model_input.attn_metadata is not None
+            assert model_input.input_tokens is not None
+            assert model_input.lora_ids is not None
             self.set_active_loras(model_input.lora_requests,
```
Comment (vivekgoe) [inline vllm/worker/habana_model_runner.py:1876]: Consider separating out mask creating code into its own code-block, will make it easier to compare with model_runner code in future.

File: vllm/worker/habana_model_runner.py [hunk 4]
```diff
@@ -1853,6 +1793,65 @@ def _check_config(self, batch_size, seq_len, is_prompt, warmup_mode):
             logger.warning("Configuration: (%s, %s, %s) was not warmed-up!",
                            phase, batch_size, seq_len)
 
+    def create_lora_mask(self, input_tokens: torch.Tensor, lora_ids: List[int],
+                         is_prompt: bool):
```
Comment (vivekgoe) [inline vllm/worker/habana_model_runner.py:1797]: Please add a short help string describing 
-  Why are we creating lora mask?
-  Why we have different masks for prompt, decode, logits?
-  How are we creating this mask?

**Review by vivekgoe — CHANGES_REQUESTED** [review]
Please also check if these changes do not break long lora context test. We may need to port these changes to long lora context branch to check that (maybe Sanju/Ruheena can do this check).

File: vllm/worker/habana_model_runner.py (hunk 2 above)
Comment (hlahkar) [reply vllm/worker/habana_model_runner.py]: These are deleted

**Review by vivekgoe — APPROVED** [review]
Looks good to me.
